internal/controller/status.go

```diff
@@ -118,13 +118,15 @@ func isIdle(ctx context.Context, clnt metricsv1beta1.PodMetricsesGetter, cr *ama
 
 	cpuUsage := metrics.Cpu()
 	if cpuUsage != nil && cpuUsage.Cmp(cpuUsageIdlenessThreshold) == -1 {
-		log.Info(
-			"the session was found to be idle",
-			"cpu usage milicores",
-			cpuUsage.MilliValue(),
-			"idle threshold milicores",
-			cpuUsageIdlenessThreshold.MilliValue(),
-		)
+		if cr.Status.IdleSince.IsZero() {
```

Comment:
question: shouldn't this be negated? If I read correctly, in children.go line 304, we set the time to Zero if the session isn't idle. So it this if would only execute when it isn't idle, but then print "session was found to be idle"? or am I misunderstanding the logic here?

Reply:
Since this check is done at a regular interval, the goal is to have the message printed only when the idleness threshold is reached the first time. Otherwise, each follow up check that would confirm the idleness would print it again.